### PR TITLE
config: pipeline: Update bookworm-rt rootfs

### PR DESCRIPTION
**Pull request by musamaanjum**
New package has been added to bookworm-rt rootfs. Hence update the URL to newly updated rootfs.

Related PR: https://github.com/kernelci/kernelci-core/pull/2614

### CODE REVIEW

**Comment by musamaanjum** [comment]
I've set the rootfs URL to staging. After verification I'll put URL of production before merging on this PR.

**Comment by a-wai** [comment]
> I've set the rootfs URL to staging. After verification I'll put URL of production before merging on this PR.

Sounds good, can you please set this PR as draft so it can't get merged as-is by accident?

File: config/pipeline.yaml
Comment (pawiecz) [inline config/pipeline.yaml:924]: Images were rebuilt to include changes from https://github.com/kernelci/kernelci-core/pull/2614, you may use here:

```suggestion
    nfsroot: 'https://storage.kernelci.org/images/rootfs/debian/bookworm-rt/20240806.0/{debarch}'
```

**Review by pawiecz — CHANGES_REQUESTED** [review]

File: config/pipeline.yaml
Comment (musamaanjum) [inline config/pipeline.yaml:924]: https://lava.collabora.dev/scheduler/job/15194120 ran on 16th August. It is using https://storage.chromeos.kernelci.org/images/rootfs/debian/bookworm-cros-flash/20240422.0/amd64/initrd.cpio.gz which is completely wrong. I'm not sure what's going on. I'll work with KernelCI team to debug it.

Comment (musamaanjum) [reply config/pipeline.yaml:924]: @pawiecz @nuclearcat please have a look here

Comment (nuclearcat) [reply config/pipeline.yaml:924]: Thats because -rt jobs defined in platforms-chromeos.yaml where they don't belong.
arm64-chromebook-device have in definition bookworm-cros-flash as all jobs here is chromeos jobs.

Comment (musamaanjum) [reply config/pipeline.yaml:924]: @nuclearcat Both builds and tests for rt are defined in pipeline.yaml (a fix has been merged last or last to last week). What are you referring to in platforms-chromeos?

**Comment by nuclearcat** [comment]
LAVA job referenced in link is from production, which is not updated yet

**Comment by musamaanjum** [comment]
The test failed in reality as `rtla` isn't found. The test result reporting is wrong. Let's debug why failure happend:

```
kernelmsg (rt-tests-rtla-osnoise)
KernelCI node: [66bf60458ac59a075e697815](https://staging.kernelci.org:9000/viewer?node_id=66bf60458ac59a075e697815)
Result: PASS
Date: 2024-08-16T14:20:53.251000
Tree: stable-rt
Branch: v6.1-rt
Kernel version: v6.1.105-rt38
Commit: 8fe1d58eff313241a0c3187db014fbccaad0b680 (https://git.kernel.org/pub/scm/linux/kernel/git/rt/linux-stable-rt.git)
Arch : arm
Platform : imx6q-sabrelite
Config: multi_v7_defconfig+preempt_rt
Compiler: gcc-12
Runtime: lava-collabora
Logs:
[lava_log](https://kciapistagingstorage1.file.core.windows.net/staging/rt-tests-rtla-osnoise-66bf5f908ac59a075e697801/log.txt.gz?sv=2022-11-02&ss=f&srt=sco&sp=r&se=2024-10-17T19:19:12Z&st=2023-10-17T11:19:12Z&spr=https&sig=sLmFlvZHXRrZsSGubsDUIvTiv%2BtzgDq6vALfkrtWnv8%3D)
```

From above node link, I've found the grand parent node (https://staging.kernelci.org:9000/viewer?node_id=66bf5f908ac59a075e697801) to find lava log. 
https://kciapistagingstorage1.file.core.windows.net/staging/rt-tests-rtla-osnoise-66bf5f908ac59a075e697801/lava_callback.json.gz?sv=2022-11-02&ss=f&srt=sco&sp=r&se=2024-10-17T19:19:12Z&st=2023-10-17T11:19:12Z&spr=https&sig=sLmFlvZHXRrZsSGubsDUIvTiv%2BtzgDq6vALfkrtWnv8%3D

This log shows that older and wrong kselftest rootfs is being used:
```
name: rt-tests-rtla-osnoise
storage.kernelci.org/images/rootfs/debian/bookworm-kselftest/20240221.0
```

Why the bookworm-rt isn't being used here? @nuclearcat @pawiecz please can you help debug this?

**Comment by pawiecz** [comment]
@musamaanjum It looks like only [depthcharge boot method currently supports `nfsroot` substitution](https://github.com/kernelci/kernelci-core/blob/a7ce4ea5626ce423960a46b85825397003bbfad4/config/runtime/boot/depthcharge.jinja2#L27). [Boot methods for other bootloaders have hardcoded images in their templates](https://github.com/kernelci/kernelci-core/tree/main/config/runtime/boot), in this case [`u-boot` template](https://github.com/kernelci/kernelci-core/blob/a7ce4ea5626ce423960a46b85825397003bbfad4/config/runtime/boot/u-boot.jinja2#L12) needs to be updated.

**Comment by musamaanjum** [comment]
> @musamaanjum It looks like only [depthcharge boot method currently supports `nfsroot` substitution](https://github.com/kernelci/kernelci-core/blob/a7ce4ea5626ce423960a46b85825397003bbfad4/config/runtime/boot/depthcharge.jinja2#L27). [Boot methods for other bootloaders have hardcoded images in their templates](https://github.com/kernelci/kernelci-core/tree/main/config/runtime/boot), in this case [`u-boot` template](https://github.com/kernelci/kernelci-core/blob/a7ce4ea5626ce423960a46b85825397003bbfad4/config/runtime/boot/u-boot.jinja2#L12) needs to be updated.

Many thanks @pawiecz. I was scratching my head over it from some time.

@nuclearcat @a-wai can we schedule the work to update the templates? You may have pre-planned it already. We should prioritize fixing rt-tests work as the work started on the rt maintainer's request and we don't want to lose his interest. (cc: @helen-fornazier as she is leading the community engagement work and can help in deciding the priority of this work.)

**Comment by a-wai** [comment]
@musamaanjum this was trivial enough that I pushed https://github.com/kernelci/kernelci-core/pull/2641

**Comment by a-wai** [comment]
Setting staging-skip to allow testing https://github.com/kernelci/kernelci-pipeline/pull/766

**Comment by pawiecz** [comment]
#766 has been merged, dropping `staging-skip` label to bring this PR back to staging

**Comment by a-wai** [comment]
Note: with #766 merged it now conflicts, so this also needs a rebase to be deployed.

**Comment by musamaanjum** [comment]
@a-wai Thanks for fixing. Rebased the PR.

**Review by pawiecz — APPROVED** [review]
LGTM

[Most recently triggered Maestro Jobs (bottom)](https://staging.kernelci.org:9000/dashboard?treetests=8fe1d58eff313241a0c3187db014fbccaad0b680) showed that [the LAVA TestJob definition might still need some adjustments](https://lava.collabora.dev/scheduler/job/15335401#L1085) but the rootfs now contains necessary tools which should be enough to merge this PR as well as https://github.com/kernelci/kernelci-core/pull/2614

**Comment by musamaanjum** [comment]
I've collected the results on https://github.com/kernelci/kernelci-project/issues/439 and fix issues under this ticket. Overall, 3/11 tests have issues in their executor script in https://github.com/Linaro/test-definitions.

**Pull request merged**
